### ISSUE TITLE
memory leak triggered by scanner

### ISSUE THREAD

**Issue by Crunsher** [Icinga/icinga2#7244]
## Describe the bug

This bug was preceded by a scan from Nessus, following this the InfluxdbWriter reports
```
[2019-06-11 14:30:31 +0000] warning/InfluxdbWriter: TLS handshake with host 'host' failed.
[2019-06-11 14:30:31 +0000] warning/InfluxDbWriter: Flush failed, cannot connect to InfluxDB: Error: std::exception
```

After the memory usage increases until the issues with memory allocation occur and the InfluxdbWriter fills the log with
```
[2019-06-11 20:12:43 +0000] critical/Socket: accept() failed with error code 24, "Too many open files"
[2019-06-11 20:12:43 +0000] critical/ApiListener: Cannot accept new connection.
```
at a rate of 3 MB/s

## To Reproduce

The bug could not be reliably reproduced.

## Expected behavior

Icinga 2 is resistant to scanners, no memory leak and in case of no more memory no log spam

## Your Environment
Icinga2 version: r2.10.5-1 Installed via Puppet, that adds the following apt source: deb http://packages.icinga.com/ubuntu icinga-bionic main Platform: Ubuntu Platform version: 18.04.2 LTS (Bionic Beaver) Kernel: Linux Kernel version: 4.15.0-38-generic Architecture: x86_64 Features enabled on master: api, checker, command, influxdb, mainlog, notification

## Additional context

This is not the first time Nessus causes issues with Icinga2. See #6559 
With the the upcoming 

ref/IP/14971

**Comment by dnsmichi** [comment]
Snapshot packages are available for testing - https://icinga.com/docs/icinga2/snapshot/doc/21-development/#snapshot-packages-nightly-builds

**Comment by dnsmichi** [comment]
I'm closing this since it is a known problem being addressed with 2.11. An RC is scheduled for end of July if you don't want to test the snapshot packages just yet.
